### PR TITLE
Fix auto replacement of deprecated args for schema inheritance

### PR DESCRIPTION
**Pull request by klecki**
The deprecated arguments handling didn't look up parent
schemas recursively.

Added test for replacing deprecated arguments in parent schema.

Replaced some deprecated readers usage.

Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

#### Why we need this PR?
Fix bug of not handling deprecation recursively in schemas. 

#### What happened in this PR? 
 - What solution was applied:
     Look up arg deprecation recursively. 
 - Affected modules and functionalities:
     Schema and spec
 - Key points relevant for the review:
     Implementation?
 - Validation and testing:
     Backend tests added
 - Documentation (including examples):
     Some unrelated readers fixes. 

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Review by JanuszL — APPROVED** [review]

File: dali/pipeline/operator/op_schema.cc [hunk 1]
```diff
@@ -111,6 +111,33 @@ std::map<std::string, DefaultedArgumentDef> OpSchema::GetOptionalArguments() con
...
+  auto it = deprecated_arguments_.find(arg_name);
+  if (it != deprecated_arguments_.end()) {
+    return it->second;
+  }
+  for (const auto &parent_name : parents_) {
+    const OpSchema &parent = SchemaRegistry::GetSchema(parent_name);
+    if (parent.IsDeprecatedArg(arg_name))
+      return parent.DeprecatedArgMeta(arg_name);
+  }
+  DALI_ENFORCE(IsDeprecatedArg(arg_name),
```
Comment (jantonguirao) [inline dali/pipeline/operator/op_schema.cc:135]: Either enforce or throw.

File: dali/pipeline/operator/op_schema.cc [hunk 2]
```diff
@@ -111,6 +111,33 @@ std::map<std::string, DefaultedArgumentDef> OpSchema::GetOptionalArguments() con
...
+               make_string("Argument \"", arg_name, "\" is not marked as a deprecated argument"));
+  throw std::runtime_error(
```
Comment (jantonguirao) [inline dali/pipeline/operator/op_schema.cc:137]: DALI_FAIL?

File: dali/pipeline/operator/op_schema.cc (hunk 1 above)
Comment (jantonguirao) [reply dali/pipeline/operator/op_schema.cc:135]: I think the enforce is redundant. If you reached here you know that IsDeprecatedArg would return false.

Comment (klecki) [reply dali/pipeline/operator/op_schema.cc]: done

File: dali/pipeline/operator/op_schema.cc (hunk 2 above)
Comment (klecki) [reply dali/pipeline/operator/op_schema.cc]: done

**Comment by klecki** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**